osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/networking/services/commands/Iec61850SetConfigurationCommand.java

```diff
@@ -39,6 +38,12 @@
     private static final int SWITCH_TYPE_TARIFF = 0;
     private static final int SWITCH_TYPE_LIGHT = 1;
 
+    private DeviceMessageLoggingService loggingService;
+
+    public Iec61850SetConfigurationCommand(final DeviceMessageLoggingService loggingService) {
+        this.loggingService = loggingService;
+    }
+
     public void setConfigurationOnDevice(final Iec61850Client iec61850Client, final DeviceConnection deviceConnection,
             final ConfigurationDto configuration) throws ProtocolAdapterException {
         final Function<Void> function = new Function<Void>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 180 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1188)